apps/studio/src/server/modules/resource/resource.router.ts

```diff
@@ -141,21 +149,14 @@ export const resourceRouter = router({
       }
 
       const result = await query.execute()
-      if (result.length > limit) {
-        // Dont' return the last element, it's just for checking if there are more
-        result.pop()
-        return {
-          items: result,
-          nextOffset: offset + limit,
-        }
-      }
-      return {
-        items: result,
-        nextOffset: null,
-      }
+      return result.length > limit
+        ? // Dont' return the last element, it's just for checking if there are more
+          { items: result.slice(0, limit), nextOffset: offset + limit }
+        : { items: result, nextOffset: null }
```

Comment:
comment/nit: this seems like unnecessary optimisation (actually it's less performant so not really an optimisation too (`slice` recreates the array, `pop` mutates)) + less readable. Suggest sticking with previous implementation.